src/file_dialog.rs

```diff
@@ -1129,6 +1128,33 @@ impl FileDialog {
         if !is_open {
             self.cancel();
         }
+
+        let mut new_file_dropped = false;
+
+        // Collect dropped files:
+        ctx.input(|i| {
+            if !i.raw.dropped_files.is_empty() {
+                // check if files were dropped
+                let dropped_files = &i.raw.dropped_files.clone();
+                if let Some(dropped_file) = dropped_files.last() {
+                    if let Some(path) = &dropped_file.path {
+                        if path.is_dir() {
+                            // if we dropped a directory, go there
+                            self.load_directory(path.as_path());
+                        } else if let Some(parent) = path.parent() {
+                            // else, go to the parent directory
+                            self.load_directory(parent);
+                        }
+                        new_file_dropped = true;
+                    }
+                }
+            }
+        });
+
+        // update GUI if we dropped a file
+        if new_file_dropped {
+            ctx.request_repaint();
+        }
```

Comment:
This can be reduced:
```suggestion
        // Collect dropped files:
        ctx.input(|i| {
            if let Some(dropped_file) = i.raw.dropped_files.last() {
                if let Some(path) = &dropped_file.path {
                    if path.is_dir() {
                        // if we dropped a directory, go there
                        self.load_directory(path.as_path());
                        ctx.request_repaint();
                    } else if let Some(parent) = path.parent() {
                        // else, go to the parent directory
                        self.load_directory(parent);
                        ctx.request_repaint();
                    }
                }
            }
        });
```

Could you also move this to a separate method like `update_dropped_files`?

Reply:
I tried this before, but somehow the whole application hangs if i call `ctx.request_repaint()` inside of `ctx.input( /*...*/)`. That's why I had to move it out

Reply:
Ah ye sry, that's the deadlock using `ctx.input()`. See: https://docs.rs/egui/latest/egui/struct.Context.html#method.input

Then you can keep the repaint separated.

But I've also thrown out a few ifs in the suggestion.

Reply:
Yes, thank, I implemented them.